### ISSUE TITLE
Travis can't push Documenter documentation to github

### ISSUE THREAD

**Issue by richardreeve** [JuliaDocs/Documenter.jl#237]
I've tried to follow your [instructions](https://juliadocs.github.io/Documenter.jl/latest/man/hosting.html#SSH-Deploy-Keys-1) to get Travis to automatically update my documentation on github for my [Diversity](https://github.com/richardreeve/Diversity.jl) package. I can generate the docs on my laptop, I have an (old) [gh-pages](https://github.com/richardreeve/Diversity.jl/tree/gh-pages) branch on github, and I have the .documenter.enc file on github, and Travis has the appropriate keys on its site. However, when I hook Documenter into Travis via the [.travis.yml](https://github.com/richardreeve/Diversity.jl/blob/master/.travis.yml) file and commit something to master, the [report](https://travis-ci.org/richardreeve/Diversity.jl/jobs/154544145) says that it generates the docs, but then doesn't have permission to push the gh-pages branch:

```
INFO    -  Building documentation to directory: /home/travis/.julia/v0.5/Diversity/doc/site 
Documenter: pushing new documentation to remote: github.com/richardreeve/Diversity.jl.git:gh-pages.
Agent pid 3817
Identity added: /home/travis/.julia/v0.5/Diversity/doc/.documenter (/home/travis/.julia/v0.5/Diversity/doc/.documenter)
Initialized empty Git repository in /tmp/tmp6i91w7/.git/
Warning: Permanently added the RSA host key for IP address '192.30.253.113' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.
```

Any suggestions would be welcome...

EDIT: I appreciate this is unlikely to be directly a Documenter issue, but I have tried to follow Documenter's own online docs, without any noticeable success...

**Comment by MichaelHatherly** [comment]
Have you copied the private key displayed by `Travis.genkeys` to your GitHub repo settings page?

**Comment by richardreeve** [comment]
No, I don't see that in the docs... is there any way of doing it retrospectively?

**Comment by MichaelHatherly** [comment]
Have you still got the terminal open that you called `genkeys` in?

**Comment by MichaelHatherly** [comment]
If not then you'll just need to redo the `genkeys` step again and replace the `.documenter.enc` file.

**Comment by richardreeve** [comment]
Ah, no. I did it a while ago in a branch, and I thought it wasn't working because I wasn't in master. I thought it was a bit odd that just running something in Julia gave Travis access to github, but it did seem to be connecting somehow to github, so I figured it was all fine.

I've now redone this. Note to self: _must read output from running commands in REPL_. Hopefully everything will now work. It would probably help for idiots like me to mention in the docs that there is stuff to do after running Travis.genkeys().

**Comment by richardreeve** [comment]
Unfortunately, this still fails. Im sure I'm still doing something stupid. The [report](https://travis-ci.org/richardreeve/Diversity.jl/jobs/154730927) says:

```
INFO    -  Building documentation to directory: /home/travis/.julia/v0.5/Diversity/doc/site 
Documenter: pushing new documentation to remote: github.com/richardreeve/Diversity.jl.git:gh-pages.
ERROR: LoadError: no keys found in ENV 'key/iv' pair.
 in getenv(::Regex) at /home/travis/.julia/v0.5/Documenter/src/Documenter.jl:416
[...]
```

The two keys do show up on the Travis settings page, and the private key, that I've called "Travis" is on github now, and claims it has never been used. Any other thoughts?

By the way, Documenter also gives slightly incorrect information on what to do to the .travis.yml file:
```
Please add the following to your build script (before_install stage in your .travis.yml, for instance):

    openssl aes-256-cbc -K $encrypted_b827f4fd7e05_key -iv $encrypted_b827f4fd7e05_iv -in .documenter.enc -out .documenter -d
```
These should presumably be `-in docs/.documenter.enc -out docs/.documenter` with the correct path as otherwise travis fails there.

**Comment by MichaelHatherly** [comment]
> with the correct path as otherwise travis fails there.

From the docs:

> If you see a line about adding an `openssl` command to your `.travis.yml` file, you can ignore that (Documenter will handle that for you).

Don't add that `openssl` line to the travis file.

**Comment by MichaelHatherly** [comment]
> mention in the docs that there is stuff to do after running Travis.genkeys().

I'll make things a little clearer in the docs later today or tomorrow if I've got some time.

**Comment by MichaelHatherly** [comment]
Also, you'll need to remove the old keys

```
encrypted_e4cbbc011ad3_key
encrypted_e4cbbc011ad3_iv
```

from the travis settings page as well.

**Comment by richardreeve** [comment]
Okay, great, that now works! Sorry for wasting your time, but I don't know anything about travis and so this was all just randomly hitting things as far as I was concerned!

**Comment by MichaelHatherly** [comment]
> Okay, great, that now works! Sorry for wasting your time, but I don't know anything about travis and so this was all just randomly hitting things as far as I was concerned!

No worries, you're not wasting my time, but rather saving time for the next person to try this.

I'll keep this open as a doc issue to remind me to rewrite that portion of the docs.

**Comment by richardreeve** [comment]
Great. I've now gone back through the docs with a view to putting in a PR, but it seems a lot clearer now I know what it's saying! I think I just got confused between adding the .documenter.enc file to my repository and adding the public key to the github setting page (and not adding the openssl bit, which I got right the first time around!). I'm not sure how to clear that up as you mention both and they are quite distinct in retrospect...

**Comment by MichaelHatherly** [comment]
Probably the most useful changes to be made in that section would be to:

- remove the discussion about GitHub "tokens" completely and only mention the SSH key approach since the tokens bit probably just clutters things up;
- add an example of the output generated by `Travis.genkeys` to make it a bit easier to explain what part needs to be copied to GitHub and which can be ignored.

**Comment by richardreeve** [comment]
I had a go at this from the output from my `Travis.genkeys()` call. Feel free to ignore it if it's not what you're looking for.

**Comment by MichaelHatherly** [comment]
Closed by #238.
